crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -516,16 +516,28 @@ impl<'w> EntityMut<'w> {
     /// Gets mutable access to the component of type `T` for the current entity.
     /// Returns `None` if the entity does not have a component of type `T`.
     #[inline]
-    pub fn get_mut<T: Component>(&mut self) -> Option<Mut<'_, T>> {
+    pub fn get_mut<T: ComponentMut>(&mut self) -> Option<Mut<'_, T>> {
         // SAFETY: &mut self implies exclusive access for duration of returned value
         unsafe { self.0.get_mut() }
     }
 
+    /// Gets mutable access to the component of type `T` for the current entity.
+    /// Returns `None` if the entity does not have a component of type `T`.
+    ///
+    /// # Safety
+    ///
+    /// - `T` must be a mutable component
+    #[inline]
+    pub unsafe fn get_mut_assume_mutable<T: Component>(&mut self) -> Option<Mut<'_, T>> {
```

Comment:
This needs a dynamic equivalent.

Reply:
The `get_mut_by_id` method is able to perform the check at runtime, since it already needs the `ComponentInfo` to find it. I _could_ add an alternate method to skip that single branch, but I'd rather not duplicate more APIs if it can be avoided. If users need it they'll ask for it.

Reply:
Aha, got it :)